codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -124,12 +141,25 @@ protected void decode(final ChannelHandlerContext ctx, HttpObject msg, List<Obje
             HttpMessage m = (HttpMessage) msg;
 
             // Handle the 'Expect: 100-continue' header if necessary.
-            // TODO: Respond with 413 Request Entity Too Large
-            //   and discard the traffic or close the connection.
-            //       No need to notify the upstream handlers - just log.
-            //       If decoding a response, just throw an exception.
             if (is100ContinueExpected(m)) {
-                ctx.writeAndFlush(CONTINUE).addListener(new ChannelFutureListener() {
+                if (HttpHeaders.getContentLength(m, 0) > maxContentLength) {
+                    tooLongFrameFound = true;
+                    final ChannelFuture future = ctx.writeAndFlush(EXPECTATION_FAILED.duplicate().retain());
+                    future.addListener(new ChannelFutureListener() {
+                        @Override
+                        public void operationComplete(ChannelFuture future) throws Exception {
+                            if (!future.isSuccess()) {
+                                ctx.fireExceptionCaught(future.cause());
+                            }
+                        }
+                    });
+                    if (closeOnExpectationFailed()) {
+                        future.addListener(ChannelFutureListener.CLOSE);
+                    }
+                    ctx.pipeline().fireUserEventTriggered(HttpExpectationFailedEvent.INSTANCE);
```

Comment:
why `ctx.pipeline().fireUserEventTriggered(...)` and not `ctx fireUserEventTriggered(...)` ?